includes/discovery/vlans/q-bridge-mib.inc.php

```diff
@@ -33,43 +33,28 @@
 
     $vtpdomain_id = '1';
 
-    //get vlan names
-    $vlans = snmpwalk_cache_oid($device, 'dot1qVlanStaticName', [], 'Q-BRIDGE-MIB');
-
-    //iterate through 'current' then 'static' dot1qVlan table
-    foreach (['Current', 'Static'] as $table) {
-        $oidsu = trim(snmp_walk($device, 'dot1qVlan' . $table . 'UntaggedPorts', ['-OsQ', '--hexOutputLength=0'], 'Q-BRIDGE-MIB'));
-        $oidsm = trim(snmp_walk($device, 'dot1qVlan' . $table . 'EgressPorts', ['-OsQ', '--hexOutputLength=0'], 'Q-BRIDGE-MIB'));
-
-        //vlan untagged ports
-        if ($oidsu) {
-            foreach (explode("\n", $oidsu) as $datau) {
-                $dots = max(array_keys(explode('.', $datau))); //last dot is index
-                $index = trim(explode('=', explode('.', $datau)[$dots])[0]); //vlan index
-                $portsu[$index] = trim(explode('=', $datau)[1]); //untagged ports
+    // fetch vlan data
+    $vlans = SnmpQuery::walk('Q-BRIDGE-MIB::dot1qVlanCurrentUntaggedPorts')->table(2);
```

Comment:
shouln't we make walk accept an array?

Reply:
@Jellyfrog Maybe.  It is tricky because snmpwalk only accepts one OID, so accepting an array of OIDs is only cosmetic and may not combine the data correctly.  Like here for example, Two of the 5 OIDs has a different index.